src/components/hooks/useSetState.tsx

```diff
@@ -0,0 +1,12 @@
+import React, {useReducer} from 'react'
+
+function useSetState(initialState:unknown) {
```

Comment:
Where's the super nice typed generic that we found? :)

Reply:
Isn't pushed yet. Since I'm waiting you to finish reviewing the code , Then I'll push everything fixed.

Reply:
You should push stuff while reviewing or else I won't see the changes.